pkg/deploy/service_accounts.go

```diff
@@ -49,43 +56,26 @@ func (s *ServiceAccounts) fetchServiceAccount(nsName string, saName string) (str
 		return "", fmt.Errorf("Internal inconsistency: Expected service account name to not be empty")
 	}
 
-	sa, err := s.coreClient.CoreV1().ServiceAccounts(nsName).Get(context.Background(), saName, metav1.GetOptions{})
+	tokenMgr := token.NewManager(s.coreClient, s.log)
```

Comment:
since NewManager is made every time we call Find no caching happens afaik.

Reply:
yup, I am wondering where it makes sense to have this, perhaps all the way back up the call stack.

Both AppCR and PKGR called `deploy.NewFactory()` so they would both get a different cache, I think having a shared cache is fine.